x-pack/test/functional/apps/ml/anomaly_detection/anomaly_explorer.ts

```diff
@@ -126,6 +137,154 @@ export default function ({ getService }: FtrProviderContext) {
+          await ml.anomaliesTable.assertTableRowsCount(25);
+        });
+
+        it('allows to change the swim lane pagination', async () => {
+          await ml.testExecution.logTestStep('checks default pagination');
+          await ml.swimLane.assertPageSize(viewBySwimLaneTestSubj, 10);
+          await ml.swimLane.assertActivePage(viewBySwimLaneTestSubj, 1);
+
+          await ml.testExecution.logTestStep('updates pagination');
+          await ml.swimLane.setPageSize(viewBySwimLaneTestSubj, 5);
+
+          const axisLabels = await ml.swimLane.getAxisLabels(viewBySwimLaneTestSubj, 'y');
+          expect(axisLabels.length).to.eql(5);
+
+          await ml.swimLane.selectPage(viewBySwimLaneTestSubj, 3);
+
+          await ml.testExecution.logTestStep('resets pagination');
+          await ml.swimLane.setPageSize(viewBySwimLaneTestSubj, 10);
+          await ml.swimLane.assertActivePage(viewBySwimLaneTestSubj, 1);
+        });
+
+        it('support cell selection by click on View By swim lane', async () => {
+          await ml.testExecution.logTestStep('clicks on the View By swim lane cell');
+          await ml.anomalyExplorer.assertSwimlaneViewByExists();
+          const sampleCell = (await ml.swimLane.getCells(viewBySwimLaneTestSubj))[0];
+
+          await ml.swimLane.selectSingleCell(viewBySwimLaneTestSubj, {
+            x: sampleCell.x + cellSize,
+            y: sampleCell.y + cellSize,
+          });
+
+          await ml.swimLane.assertSelection(viewBySwimLaneTestSubj, {
+            x: [1454817600000, 1454832000000],
+            y: ['AAL'],
+          });
+
+          await ml.testExecution.logTestStep('highlights the Overall swim lane');
+          await ml.swimLane.assertSelection(overallSwimLaneTestSubj, {
+            x: [1454817600000, 1454832000000],
+            y: ['Overall'],
+          });
+
+          await ml.testExecution.logTestStep('clears the selection');
+          await ml.anomalyExplorer.clearSwimLaneSelection();
+        });
+
+        it('supports cell selection by brush action', async () => {
+          await ml.anomalyExplorer.assertSwimlaneViewByExists();
+          const cells = await ml.swimLane.getCells(viewBySwimLaneTestSubj);
+
+          const sampleCell1 = cells[0];
+          // Get cell from another row
+          const sampleCell2 = cells.find((c) => c.y !== sampleCell1.y);
+
+          await ml.swimLane.selectCells(viewBySwimLaneTestSubj, {
+            x1: sampleCell1.x + cellSize,
+            y1: sampleCell1.y + cellSize,
+            x2: sampleCell2!.x + cellSize,
+            y2: sampleCell2!.y + cellSize,
+          });
+
+          await ml.swimLane.assertSelection(viewBySwimLaneTestSubj, {
+            x: [1454817600000, 1454846400000],
+            y: ['AAL', 'VRD'],
+          });
+
+          await ml.testExecution.logTestStep('clears the selection');
+          await ml.anomalyExplorer.clearSwimLaneSelection();
```

Comment:
Would be good to do some additional checks similar to what's done above for the overall swim lane: check that charts are displayed, that influencers list and anomalies list changed and the the URL state changed. And the same after clearSwimLaneSelection.